src/components/Modal.jsx

```diff
@@ -0,0 +1,158 @@
+import { css } from "@emotion/react";
+import styled from "@emotion/styled";
+import PropTypes from "prop-types";
+import { useState } from "react";
+import { useRecoilState } from "recoil";
+import { userStoredList } from "../atoms";
+
+const ModalContainer = styled.div`
+  position: fixed;
+  background-color: rgba(0, 0, 0, 0.38);
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  z-index: 101;
+`;
+
+const ModalBox = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  width: 84vw;
+  min-width: 300px;
+  max-width: 400px;
+  background-color: white;
+  padding: 1rem;
+  padding-top: 2rem;
+`;
+const Row = styled.div`
+  margin-bottom: 1rem;
+`;
+const Label = styled.label`
+  color: #b2b2b2;
+  font-weight: 500;
+`;
+const Input = styled.input`
+  display: block;
+  width: 100%;
+  margin: 0.2rem 0 0;
+  padding: 0.6rem 0.3rem;
+  box-sizing: border-box;
+  border: 1px solid #aaa;
+  border-radius: 5px;
+  outline: none;
+`;
+const Value = styled.p`
+  margin: 0.2rem 0 0;
+`;
+const ButtonWrap = styled.div`
+  display: flex;
+  width: 100%;
+  justify-content: space-between;
+  align-items: center;
+  gap: 0 1rem;
+`;
+
+const Button = styled.button`
+  cursor: pointer;
+  border-radius: 0.28rem;
+  border: 0;
+  outline: none;
+  color: white;
+  font-weight: bold;
+  font-size: 1rem;
+  flex: 1;
+  padding: 0.6rem;
+  ${({ color }) => {
+    return css`
+      background-color: ${color === "red" ? "#E74C3C" : "#3FC176"};
+      &:hover {
+        background-color: ${color === "red" ? "#df4433" : "#37b86c"};
+      }
+    `;
+  }}
+`;
+
+const InputRow = ({ labelName, value, allowEdit, onChange }) => (
+  <Row>
+    <Label>{labelName}</Label>
+    {allowEdit ? (
+      <Input
+        value={value}
+        placeholder="내용을 입력해주세요. "
+        onChange={onChange}
+      ></Input>
+    ) : (
+      <Value>{value}</Value>
+    )}
+  </Row>
+);
+
+InputRow.propTypes = {
+  labelName: PropTypes.string,
+  value: PropTypes.string,
+  allowEdit: PropTypes.bool,
+};
+
+function Modal({ type, cardData }) {
+  const [modalOpen, setModalOpen] = useState(true);
+  const [memo, setMemo] = useState(cardData.memo);
+  const [userList, setUserList] = useRecoilState(userStoredList);
+  const closeModal = () => setModalOpen(false);
+  const preventClose = (e) => e.stopPropagation();
+  const changeInput = (e) => setMemo(e.target.value);
+
+  const saveData = () => {
+    if (memo === "") return; // toast: "메모를 입력해 주세요."
+    const list = userList.map((item) =>
+      item.id === cardData.id ? { ...item, memo: memo } : item
+    );
+    setUserList(list);
+    localStorage.setItem("userList", list);
```

Comment:
`Recommend-Must`
추후 회의때 나온 트리구조에 맞춰 constants/localStorage.js 에서 값을 읽어오면 좋을 것 같습니다.
```jsx
import { USER_STORED_LIST } from '../constants/localStorage'
```
```suggestion
    localStorage.setItem(USER_STORED_LIST, list);
```